airflow/ui/src/layouts/Details/Grid/Grid.tsx

```diff
@@ -70,9 +72,41 @@ export const Grid = () => {
     [structure?.nodes, openGroupIds],
   );
 
+  // return all groupIds with children in the tree for expanding/collapsing functionality
+  const allGroupIds = useMemo(() => {
+    const traverse = (nodes: Array<NodeResponse>): Array<string> =>
+      nodes.flatMap((node) => (node.children ? [node.id, ...traverse(node.children)] : []));
+
+    return traverse(structure?.nodes ?? []);
+  }, [structure?.nodes]);
```

Comment:
Actually, we can move this outside of the component (In this case on line 39), remove the useMemo, and just pass `nodes` as a parameter.